### PR TITLE
Pin 1796 1857

### PR DESCRIPTION
**Pull request by beetlecrunch**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/notifier/service/converters/package.scala [hunk 1]
```diff
@@ -22,4 +24,11 @@ package object converters {
     type EventType = Value
     val ADDED, CREATED, DELETED, UPDATED, SUSPENDED, DEACTIVATED, ACTIVATED, ARCHIVED, WAITING_FOR_APPROVAL = Value
   }
+
+  def createMessageId(
+    dynamoService: DynamoService
+  )(resourceId: UUID)(implicit ec: ExecutionContext, contexts: Seq[(String, String)]): Future[MessageId] = for {
+    found          <- dynamoService.getOrganizationId(resourceId)
+    organizationId <- found.toFuture(new RuntimeException)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/notifier/service/converters/package.scala]: ⚠️

File: src/main/scala/it/pagopa/interop/notifier/service/converters/package.scala [hunk 2]
```diff
@@ -22,4 +24,11 @@ package object converters {
     type EventType = Value
     val ADDED, CREATED, DELETED, UPDATED, SUSPENDED, DEACTIVATED, ACTIVATED, ARCHIVED, WAITING_FOR_APPROVAL = Value
   }
+
+  def createMessageId(
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/notifier/service/converters/package.scala]:
```suggestion
  def getMessageId(
```
as it does not create an object through `DynamoService`

File: src/main/scala/it/pagopa/interop/notifier/service/converters/AgreementEventsConverter.scala [hunk 3]
```diff
@@ -1,47 +1,67 @@
...
+    case AgreementActivated(a)                   => Future.successful(MessageId(a.id, a.producerId))
+    case AgreementSuspended(a)                   => Future.successful(MessageId(a.id, a.producerId))
+    case AgreementDeactivated(a)                 => Future.successful(MessageId(a.id, a.producerId))
   }
 
   def asDynamoPayload: PartialFunction[ProjectableEvent, Either[ComponentError, DynamoEventPayload]] = {
     case e: Event => getEventPayload(e)
   }
 
   private[this] def getEventPayload(event: Event): Either[ComponentError, DynamoEventPayload] = event match {
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/notifier/service/converters/AgreementEventsConverter.scala]: no need to return an `Either`, results are always `Right`

**Comment by galales** [comment]
let's remember to open an infrastructure task to add the secondary index in couldformation template

File: src/main/scala/it/pagopa/interop/notifier/service/converters/package.scala (hunk 1 above)
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/notifier/service/converters/package.scala]: ⚠️

File: project/Versions.scala [hunk 4]
```diff
@@ -25,6 +25,6 @@ object Versions {
 object PagopaVersions {
   lazy val commonsVersion             = "1.0.x-SNAPSHOT"
   lazy val purposeManagementVersion   = "1.0.x-SNAPSHOT"
-  lazy val agreementManagementVersion = "1.0.x-SNAPSHOT"
+  lazy val agreementManagementVersion = "pin-1844-SNAPSHOT"
```
Comment (TonioGela) [inline project/Versions.scala]: intentional?

Comment (beetlecrunch) [reply project/Versions.scala]: No, just a local test.
🙇

**Review by TonioGela — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**